src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
+  const parts = date.split(separator);
+
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObj[fromFormat[i]] = parts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateObj['YY'], 10);
+
+    dateObj['YYYY'] = year < 30 ? `20${dateObj['YY']}` : `19${dateObj['YY']}`;
+  }
+
+  const newSeparator = toFormat[3];
```

Comment:
Similar to the issue with `fromFormat`, the code assumes the separator is at index 3 of the `toFormat` array. This might not be correct for all formats. Ensure that the separator is correctly identified or passed separately.